### ISSUE TITLE
Equalp for hash tables wrong

### ISSUE THREAD

**Issue by kpoeck** [clasp-developers/clasp#504]
````
CL-USER> (let ((ht1 (make-hash-table :size 2))
	       (ht2 (make-hash-table :size 2)))
	   (setf (gethash 23 ht1) 42)
	   (setf (gethash 24 ht2) 25)
	   (equalp ht1 ht2))
T
CL-USER> 
````

**Comment by kpoeck** [comment]
Problem probably in:
````
bool HashTable_O::equalp(T_sp other) const {
  if (this == &(*other)) return true;
  if (!other.generalp()) return false;
  if (!gc::IsA<HashTable_sp>(other)) return false;
  HashTable_sp hto = gc::As_unsafe<HashTable_sp>(other);
  if (this->hashTableTest() != hto->hashTableTest()) return false;
  if (this->hashTableCount() != hto->hashTableCount()) return false;
  this->map_while_true( [&hto] (T_sp key, T_sp val)->bool const {
      T_sp other_value = hto->gethash(key);
      if (!cl__equalp(val,other_value)) return false;
      return true;
    }
    );
  return true;
}
````

I have a hard time understanding the reason the first return true; but dropping it does not seem to help

**Comment by kpoeck** [comment]
inside the map_while_true the return true simply means, test the other elements. Still somehow it should be signaled, if the all elements where cl__equalp or not.

Have a patch that makes map_while_true return a boolean that is then finally returned

**Comment by kpoeck** [comment]
Yes, so map_while_true has to be changed to return a bool

**Comment by kpoeck** [comment]
Should be fixed with  #507
